src/stk/image/volume.h

```diff
@@ -173,8 +182,11 @@ namespace stk
 
         float3 _origin; // Origin in world coordinates
         float3 _spacing; // Size of a voxel
+        matrix<float, 3, 3> _direction; // Cosine directions of the axes
 
         bool _contiguous;
+
+        std::map<std::string, std::string> _metadata;
```

Comment:
Things weren't playing nice, so I decided to store metadata as strings. I think in all reasonable formats metadata are stored as such, or can be casted to strings. Now we do not have pointers hanging around, and the map can just be copied when copying the volume, without sharing the metadata.

Reply:
Ah cool, less ITK is good. Still need to add some shared reference with copy-on-write thing I think, this would also make things simpler when integrating with GpuVolume. But there's no hurry, I can sort that out later.